### PR TITLE
feat: add ability to display image in toasts for artwork lists flow

### PR DESCRIPTION
**Pull request by dimatretyak**
This PR resolves [] 

### Description
Some discussion you can find [this Slack thread](https://artsy.slack.com/archives/C9SATFLUU/p1685723980898399)

### Screenshots
| Screenshot | Screenshot |
| ------------- | ------------- |
| iOS-1 | iOS-2 | 
| iOS-3 | iOS-4 |
| iOS-5 | iOS-6 |
| iOS-7 | |

### Demo
| Android | iOS |
| ------------- | ------------- |
| <video src="https://github.com/artsy/eigen/assets/3513494/cb091a2d-a8d6-4325-b621-f4000ed50cd0" />  | <video src="https://github.com/artsy/eigen/assets/3513494/354c2ac5-0a15-480d-917b-84bf009044da" /> | 

### PR Checklist

- [x] I have tested my changes on **iOS** and **Android**.
- [x] I hid my changes behind a **[feature flag]**, or they don't need one.
- [x] I have included **screenshots** or **videos**, or I have not changed the UI.
- [ ] I have added **tests**, or my changes don't require any.
- [ ] I added an **[app state migration]**, or my changes do not require one.
- [ ] I have documented any **follow-up work** that this PR will require, or it does not require any.
- [x] I have added a **changelog entry** below, or my changes do not require one.

### To the reviewers 👀

- [ ] I would like **at least one** of the reviewers to **run** this PR on the simulator or device.

<details><summary>Changelog updates</summary>

#### Cross-platform user-facing changes

-

#### iOS user-facing changes

-

#### Android user-facing changes

-

#### Dev changes

- add ability to display image in toasts for artwork lists flow - dimatretyak

</details>

Need help with something? Have a look at our [docs], or get in touch with us.

[app state migration]: ../blob/main/docs/adding_state_migrations.md
[feature flag]: ../blob/main/docs/developing_a_feature.md
[docs]: ../blob/main/docs/README.md

### CODE REVIEW

File: src/app/Components/Toast/ToastComponent.tsx [hunk 1]
```diff
@@ -95,9 +98,16 @@ export const ToastComponent = ({
   }
 
   const innerTopBottom = (
-    <Flex flex={1} flexDirection="row" alignItems="center" mx={2}>
+    <Flex flex={1} flexDirection="row" alignItems="center" mx={1}>
```
Comment (anandaroop) [inline src/app/Components/Toast/ToastComponent.tsx:101]: **question (non-blocking):** Intentional margin change?

File: src/app/Components/Toast/ToastComponent.tsx [hunk 2]
```diff
@@ -95,9 +98,16 @@ export const ToastComponent = ({
...
       {Icon !== undefined ? <Icon fill="white100" width={25} height={25} mr={1} /> : null}
 
+      {!!imageURL && (
+        <>
+          <OpaqueImageView imageURL={imageURL} width={IMAGE_SIZE} height={IMAGE_SIZE} />
```
Comment (anandaroop) [inline src/app/Components/Toast/ToastComponent.tsx:106]: **thought (non-blocking):** [Like Ryan](https://artsy.slack.com/archives/C9SATFLUU/p1686044567844899?thread_ts=1685723980.898399&cid=C9SATFLUU), I was wondering how this would look for artworks that happen to match or clash with the colors of the toasts. Should we add a 1px white border here? (Maybe we can revisit when Ryan is back)

**Review by anandaroop — APPROVED** [review]
🖼️ Very nice

File: src/app/Components/Toast/ToastComponent.tsx (hunk 2 above)
Comment (dimatretyak) [reply src/app/Components/Toast/ToastComponent.tsx:106]: > Maybe we can revisit when Ryan is back

yeah, agree 👍

**Review by nickskalkin — APPROVED** [review]

**Review by nickskalkin — DISMISSED** [review]

**Review by dblandin — APPROVED** [review]
Very cool 💯

**Comment by dimatretyak** [comment]
> I was wondering how this would look for artworks that happen to match or clash with the colors of the toasts. Should we add a 1px white border here?

Some discussion of this [here](https://artsy.slack.com/archives/C9SATFLUU/p1686241715518479)

**Pull request merged**